integration/update_field_compat_test.go

```diff
@@ -669,23 +669,30 @@ func TestUpdateFieldCompatSet(t *testing.T) {
 		},
 		"DocFieldExist": {
 			update: bson.D{{"$set", bson.D{{"v.foo", int32(1)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1661",
 		},
 		"DocumentFieldNotExist": {
 			update:        bson.D{{"$set", bson.D{{"foo.bar", int32(1)}}}},
 			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1676",
 		},
 		"ArrayFieldExist": {
-			update: bson.D{{"$set", bson.D{{"v.array.0", int32(1)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1661",
+			update:        bson.D{{"$set", bson.D{{"v.array.0", int32(1)}}}},
+			skipForTigris: "Tigris does not support language keyword 'array' as field name",
 		},
 		"ArrayFieldNotExist": {
 			update:        bson.D{{"$set", bson.D{{"foo.0.baz", int32(1)}}}},
 			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1676",
 		},
-		"DocArrFieldNotExist": {
-			update: bson.D{{"$set", bson.D{{"v.0.foo", int32(1)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1661",
+		"DocArrFieldNotExists_0": {
+			update:        bson.D{{"$set", bson.D{{"v.0.foo", int32(1)}}}},
+			skipForTigris: "Tigris needs a special data set: https://github.com/FerretDB/FerretDB/issues/1791",
```

Comment:
I decided not to add new Tigris data sets as part of the current task, because it slows down Tigris even more... So we need to figure out what's wrong (#1791).